### DISCUSSION TITLE
Plot of Selection efficiency vs mass

### DISCUSSION THREAD

**Discussion by ram1123** [ram1123/2l2Q_limitSettingTool#34] [Comments_HIG_23_001]
> Table 9/muon and electron isolation: For the highest masses considered, are the two leptons getting nearby so that the isolation criterion leads to a significant inefficiency? Is there a plot of selection efficiency vs mass? Have you considered using mini-isolation?

- [x] Get the plot of selection efficiency vs mass
- [x] Get the plot of lepton selection efficiency vs mass

**Comment by jialin-guo1** [comment]
**Get the plot of selection efficiency vs mass**
 
 these efficiency vs mass plot already showed on Fig.24 Fig.25

![image](https://github.com/ram1123/2l2Q_limitSettingTool/assets/68882142/b68c912a-2080-41c6-b455-a9afe42ddcc2)

**Reply by jialin-guo1** [reply]
![image](https://github.com/ram1123/2l2Q_limitSettingTool/assets/68882142/588d0bae-d2d6-40db-8648-bd169c781ef8)

 lepton selection efficiency vs mass

**Comment by ram1123** [comment]
Hi @jialin-guo1 ,

Try to plot the deltaR between two generator level leptons. For 500, 1500, 3000 GeV mass points.

**Reply by jialin-guo1** [reply]
**mass 500**:
![image](https://github.com/ram1123/2l2Q_limitSettingTool/assets/68882142/0a0330b3-b19c-4b5f-9a94-8a806c705345)

**mass 1500**
![image](https://github.com/ram1123/2l2Q_limitSettingTool/assets/68882142/56537810-6ac3-41c4-b6e2-74b595011374)

**mass 3000**
![image](https://github.com/ram1123/2l2Q_limitSettingTool/assets/68882142/265cb38b-c652-4b70-bffa-873e04078c39)

**Reply by jialin-guo1** [reply]
@ram1123 
update delatEta to deltaR **computation**: deltaR = sqrt((phi1 - phi2) * (phi1 - phi2) + (eta1 - eta2) * (eta1 - eta2))

mass 500:
![image](https://github.com/ram1123/2l2Q_limitSettingTool/assets/68882142/45f91121-1f19-41a5-96b7-a29f71824518)

mass1500:
![image](https://github.com/ram1123/2l2Q_limitSettingTool/assets/68882142/6f199518-94ed-48b8-bfa5-3498f5540f14)

mass3000:
![image](https://github.com/ram1123/2l2Q_limitSettingTool/assets/68882142/3d87039a-4770-4171-bbc4-c0b4a54f1645)

**Reply by ram1123** [reply]
I think you need to update the dPhi. Its not correct.

Whenever the dPhi is greater than pi then we need to subtract it from 2PI.

**Reply by jialin-guo1** [reply]
could you please confirm this function:

def deltaR( eta1, phi1, eta2, phi2):
    if abs(phi1-phi2)>np.pi:
        dR = math.sqrt((2.*np.pi - abs(phi1 - phi2)) * (2.*np.pi - abs(phi1 - phi2)) + (eta1 - eta2) * (eta1 - eta2))
    else:
        dR = math.sqrt((phi1 - phi2) * (phi1 - phi2) + (eta1 - eta2) * (eta1 - eta2))
    return dR

**Reply by ram1123** [reply]
yes its correct.

**Reply by jialin-guo1** [reply]
mass500:
![image](https://github.com/ram1123/2l2Q_limitSettingTool/assets/68882142/9d4ef7b2-6638-4fa7-8c53-f957d5659ecd)

**Reply by jialin-guo1** [reply]
mass 3000
<img width="728" alt="image" src="https://github.com/ram1123/2l2Q_limitSettingTool/assets/68882142/dd507782-ccd8-4fd8-9e75-c434c40577f6">
